main.go

```diff
@@ -16,21 +16,27 @@ var (
 )
 
 func main() {
+	os.Exit(Main())
+}
+
+func Main() int {
```

Comment:
Why there is 2 main / Main functions?

Reply:
To understand, it helps to keep in mind that Go is strongly typed. This makes it easy to follow the flow.

A Go program wants a main function with name and signature `main()`. That is, it takes nothing and returns nothing. (This is not the same as `None` in Python or `nil` in Go, it is really nothing).

Then we have `Main() int`, which returns an integer. This could have been called anything. We call it `Main` as convention, because it is doing everything the `main()` is doing.

Why we added it? Because of 

```go
     1  func TestMain(m *testing.M) {
     2      // The commands map holds the set of command names, each with an associated
     3      // run function which should return the code to pass to os.Exit.
     4      // When [testscript.Run] is called, these commands are installed as regular
     5      // commands in the shell path, so can be invoked with "exec".
     6      os.Exit(testscript.RunMain(m, map[string]func() int{
     7          "terravalet": Main,
     8      }))
     9  }
```

See line 7 ? it calls this `Main()` and, see line 6, it is using its return code for `os.Exit`.

The power of `testscript`, introduced in this PR, is that when it sees line 7, it also compiles the test executable. Said in another way, it allows to drive integration tests directly from Go code with the standard Go tooling (`go test`), ensuring that dependencies are built before the tests are run.

Reply:
Thank you. So if I summarize roughly, you need `Main()`, which returns an integer, because you need a return value when using `TestMain` but `main()` returns nothing?

Reply:
@iAmoric exactly.

Then we could ask ourselves why `main()` returns nothing. There are reasons, rooted in the fact that not all OSes are Unix. But this is another story :-)